src/bootstrap/bootstrap.py

```diff
@@ -81,14 +81,14 @@ def run(args, verbose=False):
 
 def stage0_data(rust_root):
     nightlies = os.path.join(rust_root, "src/stage0.txt")
+    data = {}
     with open(nightlies, 'r') as nightlies:
-        data = {}
-        for line in nightlies.read().split("\n"):
-            if line.startswith("#") or line == '':
+        for line in nightlies:
+            if line.startswith("#") or line.strip() == '':
                 continue
             a, b = line.split(": ", 1)
             data[a] = b
-        return data
+    return data
```

Comment:
FYI, you can do this with a single comprehension:

```python
with open(nightlies, 'r') as nightlies:
    return dict(
        line.split(": ", 1)
        for line in nightlies
        if line and not line.strip().startswith('#')
    )
```

However, this is obviously a matter of style (and your way is probably more readable to someone who doesn't do a lot of functional programming).